code/modules/networks/computer3/bank_rec.dm

```diff
@@ -0,0 +1,1050 @@
+					if (!src.active_bank)
+						src.print_text("No bank record loaded!")
+						src.menu = MENU_IN_RECORD
+						return
+					var/newWage = round(text2num_safe(command))
+					if (newWage < 0)
+						src.print_text("<b>Error</b>: You cannot set a negative wage.")
+						return
+					if (newWage > 10000)
+						src.print_text("<b>Warning:</b> Maximum wage is 10,000[CREDIT_SIGN]")
+						newWage = 10000
+
+					src.log_string += "<br>Wage for [src.active_general["name"]] changed from [src.active_bank["wage"]][CREDIT_SIGN] to [newWage][CREDIT_SIGN]"
+					logTheThing(LOG_STATION, usr, "sets <b>[src.active_general["name"]]</b>'s wage to [newWage][CREDIT_SIGN].")
+					src.active_bank["wage"] = newWage
+
+				if(FIELDNUM_BALANCE)
+					if (inputText == "0")
+						src.print_text("Transfer cancelled.")
+						src.menu = MENU_IN_RECORD
+						return
+					if (!src.active_bank)
+						src.print_text("No bank record loaded!")
+						src.menu = MENU_IN_RECORD
+						return
+
+					var/balanceChange = round(text2num_safe(command))
+					switch(balanceChange)
+						if (-INFINITY to 0)
+							if ((src.active_bank["current_money"] + balanceChange) < 0)
+								src.print_text("<b>Warning:</b> [src.active_general["name"]] only has [src.active_bank["current_money"]][CREDIT_SIGN] in account!")
+								balanceChange = -src.active_bank["current_money"]
+							logTheThing(LOG_STATION, usr, "adds [abs(balanceChange)][CREDIT_SIGN] to the station budget from <b>[src.active_general["name"]]</b>'s account.")
+							src.print_text("Transferred [abs(balanceChange)][CREDIT_SIGN] from [src.active_general["name"]] to station budget.")
+							src.active_bank["current_money"] += balanceChange
+							global.wagesystem.station_budget -= balanceChange // balanceChange is negative here, this adds to the station budget
+						if (0 to INFINITY)
+							if (global.wagesystem.station_budget < balanceChange)
+								src.print_text("<b>Warning:</b> Station budget only has [global.wagesystem.station_budget][CREDIT_SIGN] available!")
+								balanceChange = global.wagesystem.station_budget
+							logTheThing(LOG_STATION, usr, "adds [balanceChange][CREDIT_SIGN] to <b>[src.active_general["name"]]</b>'s account from the station budget.")
+							src.print_text("Transferred [balanceChange][CREDIT_SIGN] from the station budget to [src.active_general["name"]].")
+							src.active_bank["current_money"] += balanceChange
+							global.wagesystem.station_budget -= balanceChange
+					src.menu = MENU_IN_RECORD
+				if(FIELDNUM_NOTES)
+					if (!src.active_bank)
+						src.print_text("No bank record loaded!")
+						src.menu = MENU_IN_RECORD
+						return
+					if (ckey(inputText))
+						src.active_bank["notes"] = copytext(inputText, 1, MAX_MESSAGE_LEN)
+					else
+						return
+
+				if(FIELDNUM_DELETE)
+					switch(ckey(inputText))
+						if ("y")
+							if (src.active_bank)
+								src.log_string += "<br>Bank Record [src.active_bank["id"]] deleted."
+								qdel(src.active_bank)
```

Comment:
Deleting bank records should proobably transfer the money into the payroll budget so it doesn't just vanish into thin air.

Reply:
9c645cd36b859a75b122ea85dcf13adb08047cf3